lib/discover/hyperbahn.js

```diff
@@ -0,0 +1,75 @@
+var TChannelAsThrift = require('tchannel/as/thrift');
+var TChannel = require('tchannel');
+var fs = require('fs');
+var path = require('path');
+
+module.exports.discover = function hyperbahnDiscover(urlObj, original, callback) {
+    var host;
+    if (urlObj.host) {
+        host = urlObj.host;
+    } else {
+        host = '127.0.0.1:21300';
+    }
+
+    var client = TChannel();
+    var hChannel = client.makeSubChannel({
+        serviceName: 'hyperbahn',
+        peers: [host]
+    });
+    var tchannelAsThrift = TChannelAsThrift({
+        channel: hChannel,
+        entryPoint: path.join(__dirname, 'hyperbahn.thrift')
+    });
+
+    tchannelAsThrift.waitForIdentified({
+        host: host
+    }, function onIdentified(err) {
+        if (err) {
+            callback(err);
+            return;
+        }
+
+        tchannelAsThrift.request({
+            host: host,
+            serviceName: 'hyperbahn',
+            headers: {
+                cn: 'hyperbahn'
+            },
+            hasNoParent: true
+        }).send('Hyperbahn::discover', {}, {
+            query: {
+                serviceName: urlObj.pathname.substr(1) // remove first slash
+            }
+        }, function (err, resp) {
+            if (err) {
+                return callback(err);
+            }
+
+            // parse the hyperbahn response
+            var seeds;
+            try {
+                seeds = resp.body.peers.map(function(peer) {
+                    switch (peer.ip.type) {
+                        case 'ipv4':
+                            return [intToIP(peer.ip.ipv4), peer.port].join(':')
+                        default:
+                            throw new Error("Unknown ip type '" + peer.ip.type + "'")
+                    }
+                });
+            } catch (e) {
+                return callback(e);
+            }
+
+            return callback(null, seeds);
+        });
+    });
+}
```

Comment:
We've recently ported tchannel to eslint.

We use the "extends" pattern in the eslintrc ( https://github.com/uber/tchannel-node/blob/master/.eslintrc ) and re-use our `perf-standard` rules.

You may benefit from using these rules and then overwriting them for your ringpop style preferences.